forgerock-auth/src/main/java/org/forgerock/android/auth/ConfigHelper.kt

```diff
@@ -0,0 +1,178 @@
+ * This class also verify if the configuration changes on the current and previous launch.
+ */
+class ConfigHelper: ConfigInterface {
+
+    companion object {
+
+        //Alias to store Previous Configure Host
+        const val ORG_FORGEROCK_V_1_HOSTS = "org.forgerock.v1.HOSTS"
+
+
+        /**
+         * Persist the FROption Data in shared preference
+         *
+         * @param context  The Application Context
+         * @param frOptions Argument to persist the data in shared preference using the context
+         */
+        @JvmStatic
+        fun persist(context: Context, frOptions: FROptions) {
+            val sharedPreferences =
+                context.getSharedPreferences(ORG_FORGEROCK_V_1_HOSTS, Context.MODE_PRIVATE)
+            sharedPreferences.edit().putString("url", frOptions.server.url)
+                .putString("realm", frOptions.server.realm)
+                .putString("cookieName", frOptions.server.cookieName)
+                .putString("client_id", frOptions.oauth.oauthClientId)
+                .putString("revoke_endpoint", frOptions.urlPath.revokeEndpoint)
+                .putString("end_session_endpoint", frOptions.urlPath.endSessionEndpoint)
+                .putString("scope", frOptions.oauth.oauthScope)
+                .putString("redirect_uri", frOptions.oauth.oauthRedirectUri)
+                .apply()
+        }
+
+        /**
+         * validate the configuration changes on start of the SDK
+         *
+         * @param context  The Application Context
+         * @param frOptions takes current configuration and compare with persisted configuration
+         * @return true when the config changed and false if its not changed
+         */
+        @JvmStatic
+        fun isConfigChanged(context: Context, frOptions: FROptions): Boolean {
+            val sharedPreferences =
+                context.getSharedPreferences(ORG_FORGEROCK_V_1_HOSTS, Context.MODE_PRIVATE)
+            sharedPreferences.getString("url", null)?.apply {
+                if(frOptions.server.url != this) {
+                    return true
+                }
+            }
+            sharedPreferences.getString("realm", null)?.apply {
+                if(frOptions.server.realm != this) {
+                    return true
+                }
+            }
+            sharedPreferences.getString("client_id", null)?.apply {
+                if(frOptions.oauth.oauthClientId != this) {
+                    return true
+                }
+            }
+            sharedPreferences.getString("cookieName", null)?.apply {
+                if(frOptions.server.cookieName != this) {
+                    return true
+                }
+            }
+            sharedPreferences.getString("scope", null)?.apply {
+                if(frOptions.oauth.oauthScope != this) {
+                    return true
+                }
+            }
+            sharedPreferences.getString("redirect_uri", null)?.apply {
```

Comment:
Where did we land on this?
In iOS the FROptions checks for the following:

return (lhs.url == rhs.url &&
        lhs.realm == rhs.realm &&
        lhs.cookieName == rhs.cookieName &&
        lhs.oauthClientId == rhs.oauthClientId)

do we need to align? @jeyanthanperiyasamy @witrisna @spetrov

Reply:
Sounds good. I'll raise a bug ticket for that and do it